akka-docs/rst/scala/code/docs/ddata/protobuf/TwoPhaseSetSerializer.scala

```diff
@@ -51,9 +51,11 @@ class TwoPhaseSetSerializer(val system: ExtendedActorSystem)
 
   def twoPhaseSetFromBinary(bytes: Array[Byte]): TwoPhaseSet = {
     val msg = TwoPhaseSetMessages.TwoPhaseSet.parseFrom(bytes)
-    TwoPhaseSet(
-      adds = GSet(msg.getAddsList.iterator.asScala.toSet),
-      removals = GSet(msg.getRemovalsList.iterator.asScala.toSet))
+    val addsSet = msg.getAddsList.iterator.asScala.toSet
```

Comment:
This example now matches its Java counterpart - `TwoPhaseSetSerializer.java`. However, there is interesting side-effect - doing `add` on a `GSet` accumulates deltas (this would be true also for e.g. `GCounter` in a delta-CRDT setting). I think that the right thing to do would be to `.resetDeltas` on the level where the CRDT is re-hydrated.